From 41347924799ff4f16a12d8f740f5c66b10eb8c43 Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Sun, 9 Aug 2020 20:58:52 +0200
Subject: [PATCH 05/43] scripts/ubinize-image.sh: allow setting the rootfs name

This patch allows us to use a different rootfs name. Some QCA boards will
require the in ubi rfs to be called ubi_rootfs.

Signed-off-by: John Crispin <john@phrozen.org>
---
 include/image-commands.mk | 1 +
 scripts/ubinize-image.sh  | 9 ++++++++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/include/image-commands.mk b/include/image-commands.mk
index c6e8eb0293..d4bce80e68 100644
--- a/include/image-commands.mk
+++ b/include/image-commands.mk
@@ -86,6 +86,7 @@ define Build/append-ubi
 	sh $(TOPDIR)/scripts/ubinize-image.sh \
 		$(if $(UBOOTENV_IN_UBI),--uboot-env) \
 		$(if $(KERNEL_IN_UBI),--kernel $(IMAGE_KERNEL)) \
+		$(if $(ROOTFSNAME_IN_UBI),--rootfs_name $(ROOTFSNAME_IN_UBI)) \
 		$(foreach part,$(UBINIZE_PARTS),--part $(part)) \
 		$(IMAGE_ROOTFS) \
 		$@.tmp \
diff --git a/scripts/ubinize-image.sh b/scripts/ubinize-image.sh
index c6f8bcefe5..707cd38d47 100755
--- a/scripts/ubinize-image.sh
+++ b/scripts/ubinize-image.sh
@@ -7,6 +7,7 @@ ubootenv=""
 ubinize_param=""
 kernel=""
 rootfs=""
+rootfs_name="rootfs"
 outfile=""
 err=""
 ubinize_seq=""
@@ -76,7 +77,7 @@ ubilayout() {
 		rootsize="$( round_up "$( stat -c%s "$2" )" 1024 )"
 		;;
 	esac
-	ubivol $vol_id rootfs "$2" "$autoresize" "$rootsize"
+	ubivol $vol_id $rootfs_name "$2" "$autoresize" "$rootsize"
 
 	vol_id=$(( $vol_id + 1 ))
 	[ "$rootfs_type" = "ubifs" ] || ubivol $vol_id rootfs_data "" 1
@@ -101,6 +102,12 @@ while [ "$1" ]; do
 		shift
 		continue
 		;;
+	"--rootfs_name")
+		rootfs_name="$2"
+		shift
+		shift
+		continue
+		;;
 	"--part")
 		parts="$parts $2"
 		shift
-- 
2.25.1

